packages/cells/src/widget.ts

```diff
@@ -1061,6 +1088,45 @@ export class CodeCell extends Cell<ICodeCellModel> {
     this.node.setAttribute('aria-label', ariaLabel);
   }
 
+  getOutputPlaceholderText(): string | undefined {
+    const outputData = this.model.outputs.get(0)?.data;
```

Comment:
The code below is wrong and too complex.

First the `output_type` is at the level of `this.model.outputs.get(0)`. Then depending on the output type you may have or not a `data` attributes.

So I would recommend you use a _switch-case_. In a structure like:

```
// Supported mimetype in order of preference
const supportedMimeTypes = [
  'text/html',
  'image/svg+xml',
  // ...
]
const first_output = this.model.outputs.get(0);

switch (first_output?.output_type) {
  // Treat those cases similarly
  case 'display_data':
  case 'execute_result': {
      const preferredOutput = supportedMimeTypes.find(mt => {
        const data = first_output.data[mt];
        return (Array.isArray(data) ? typeof data[0] : typeof data) === 'string'
      });
      const dataToDisplay = first_output.data[preferredOutput];
      return (Array.isArray(dataToDisplay) ? dataToDisplay : dataToDisplay.split('\n')).find(part => part !== '');
    }
  case 'stream':
    return (Array.isArray(first_output.text) ? first_output.text : first_output.text.split('\n')).find(part => part !== '');
  case 'error':
    return `{first_output.ename} {first_output.evalue}`;
}
```

Reply:
Oh ok I can definitely add a switch. Thanks!

Reply:
Ok update after looking into these changes - I'm not seeing any types of outputs that don't use the `data` field (and this is reflected in the type definition). I'm also not seeing an `ename` or an `evalue`, only the `application/vnd.jupyter.stderr` field. I agree that it should be simplified though and I pushed a change to simplify it.